src/Time.cpp

```diff
@@ -272,16 +291,66 @@ Time Time::fromString(const std::string& stringTime, Time::Resolution resolution
             "format '" + mainFormat +"'"
         );
     }
-    // " ... not set by strptime(); tells mktime() to determine
-    // whether daylight saving time is in effect ..."
-    // (http://pubs.opengroup.org/onlinepubs/007904975/functions/strptime.html)
 
-    tm.tm_isdst = -1;
-    time_t time = mktime(&tm);
+    time_t time;
+    if(tzInfo.empty())
+    {
+        tm.tm_isdst = -1;
+        time = mktime(&tm);
+    } else {
+        // full offset to UTC
+        int64_t tzOffset = Time::tzInfoToSeconds(tzInfo);
+        //// now we are in utc input
+        tm.tm_sec += tzOffset;
+
+        // mktime interpretes tm as local time
+        // We have remove the local timezone's offset (at the selected date) to UTC, to really get UTC
+        tm.tm_isdst = -1;
+        time = mktime(&tm);
```

Comment:
This whole thing is very dubious actually.

The time offset does not define the existence of DST, only the time zone name can. Algiers and Paris have the same base offset, Paris had DST in 2019, Algiers did not. 

I guess that one workable solution is to assume that the time-offset marker already includes the DST (it probably does, because of the above comment) and not handle DST at all.

Reply:
The time offset derives from '%z' strftime:
```
The +hhmm or -hhmm numeric timezone (that is, the hour and
              minute offset from UTC).
```
so DST is included.

A requirement of the implementation is, however, that setting of tm_isdst here has to be mirrored within getTimezoneOffset.
If we want getTimezoneOffset to correspond to 'timezone' for the current time, then this needs to be -1, otherwise we can fix it to 0.

Reply:
~~~
// now we move to the local timezone
~~~

Waitaminute.

Let's get back one step: do you want the Time as returned by fromString to be represented in the local time zone ?

Reply:
OK. I'm sorry if you are receiving notifications for all the comments I've left and then removed. This is as I expected, mind-numbing. Anyways. I misread the whole thing, my mind is getting tired of juggling with time zones.

I think this needs a lot more comments

~~~
else
{
  // We have to go through a few intermediate "time zones" to reach UTC
  // The problem we try to work around here is that mktime converts a broken-down time
  // in the local time zone to UTC, but what we have is a time in UTC and want to convert
  // it to UTC

  // 'tm' is the time represented by the string. It is `tzInfo` seconds away from UTC
  tm.tm_sec += tzOffset;
  // OK, now `tm` is the string's time in UTC. Note that mktime handles wraparounds
  tm.tm_isdst = -1;
  time = mktime(&tm);
  // Since mktime expected `tm` to be in local time, `time` is UTC with the local time
  // zone offset as it was at `tm` applied in opposite (e.g. time would be UTC-3 if
  // executed on a machine whose current time zone is UTC+3).
  // We have to correct with the local offset to finally get UTC
  int64_t localTimezoneOffset = Time::getTimezoneOffset(time);
  time -= localTimezoneOffset;
}
~~~

Reply:
Indeed mind-numbing, but glad you are still up for reviewing. Updated according to your suggestions here.